src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/Contexts.kt

```diff
@@ -73,10 +76,36 @@ class Contexts(
 		}
 	}
 
-	override fun closeCurrent() {
+	override fun clear() {
+		clear(true)
+	}
+
+	private fun clear(notify: Boolean) {
+		synchronized(this) {
+			all.clear()
+			closeCurrent(notify)
+		}
+	}
+
+	protected open fun refresh() {
+		clear(false)
+		observable.fireModified(this) // invalidates root bcs there's no tree node for this
+	}
+
+	private fun closeCurrent() {
+		closeCurrent(true)
+	}
+
+	private fun closeCurrent(notify: Boolean) {
+		if (current == null) {
```

Comment:
Is this check necessary? If this.current is null, its null value will be given to `val current` then the `return` will be made at `current?.close() ?: return`, no?

Reply:
great catch, thx!